src/app/shared/tabview-x/tabview-index-tracker.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Location } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TabviewIndexTrackerService {
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  private readonly FIRST_TAB = 0;
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  private readonly DEFAULT_QUERY_PARAM_KEY = 'tab';
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private location: Location,
+  ) {
+  }
+
+  /**
+   * Get the tabindex from the queryParams. if not found return index 0.
+   *
+   * @param paramKey defaults to 'tab'.
```

Comment:
nit: Would be nice to have its purpose explained instead of the default value. We could rely on the intellisense here to provide us with the default.

```suggestion
   * @param paramKey the URL's query parameter name.
```